src/CSharpLanguageServer/Server.fs

```diff
@@ -852,6 +856,36 @@ type CSharpLspServer(lspClient: CSharpLspClient, options: Options) =
 
         async { return metadataMaybe |> success }
 
+    override __.TextDocumentFormatting(format: Types.DocumentFormattingParams) : AsyncLspResult<Types.TextEdit [] option> =
```

Comment:
I rewrote request handling a bit to sequence r/w vs r/o rpc calls and you need to source `state` with `withStateRead` -- that is way it does not compile with the latest code

Reply:
Sounds good, ill fetch from upstream and make that change